Subject: MCA when shutting down tulip quad-NIC 
From: andrew.patterson@hp.com
References: SUSE39204
Patch-mainline: not yet

Shutting down the network causes an MCA because of an IO TLB error when
a DEC quad 10/100 card is in any slot.  This problem was originally seen
on an HP rx4640.

Acked-by: Olaf Kirch <okir@suse.de>

 drivers/net/ethernet/dec/tulip/tulip_core.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/ethernet/dec/tulip/tulip_core.c
+++ b/drivers/net/ethernet/dec/tulip/tulip_core.c
@@ -1940,6 +1940,10 @@ static void __devexit tulip_remove_one (
 		return;
 
 	tp = netdev_priv(dev);
+
+	/* shoot NIC in the head before deallocating descriptors */
+	pci_disable_device(tp->pdev);
+
 	unregister_netdev(dev);
 	pci_free_consistent (pdev,
 			     sizeof (struct tulip_rx_desc) * RX_RING_SIZE +
